middleware/browse/browse.go

```diff
@@ -92,189 +92,194 @@ func (l byTime) Less(i, j int) bool { return l.Items[i].ModTime.Before(l.Items[j
+                        sort.Sort(sort.Reverse(byTime(l)))
+                default:
+                        // If not one of the above, do nothing
+                        return
+                }
+        } else { // If we had more Orderings we could add them here
+                switch l.Sort {
+                case "name":
+                        sort.Sort(byName(l))
+                case "size":
+                        sort.Sort(bySize(l))
+                case "time":
+                        sort.Sort(byTime(l))
+                default:
+                        // If not one of the above, do nothing
+                        return
+                }
+        }
 }
 
 // HumanSize returns the size of the file as a human-readable string
 // in IEC format (i.e. power of 2 or base 1024).
 func (fi FileInfo) HumanSize() string {
-	return humanize.IBytes(uint64(fi.Size))
+        return humanize.IBytes(uint64(fi.Size))
 }
 
 // HumanModTime returns the modified time of the file as a human-readable string.
 func (fi FileInfo) HumanModTime(format string) string {
-	return fi.ModTime.Format(format)
+        return fi.ModTime.Format(format)
 }
 
 var IndexPages = []string{
-	"index.html",
-	"index.htm",
-	"index.txt",
-	"default.html",
-	"default.htm",
-	"default.txt",
+        "index.html",
+        "index.htm",
+        "index.txt",
+        "default.html",
+        "default.htm",
+        "default.txt",
 }
 
 func directoryListing(files []os.FileInfo, r *http.Request, canGoUp bool, root string) (Listing, error) {
-	var fileinfos []FileInfo
-	var urlPath = r.URL.Path
-	for _, f := range files {
-		name := f.Name()
-
-		// Directory is not browsable if it contains index file
-		for _, indexName := range IndexPages {
-			if name == indexName {
-				return Listing{}, errors.New("Directory contains index file, not browsable!")
-			}
-		}
-
-		if f.IsDir() {
-			name += "/"
-		}
-
-		url := url.URL{Path: name}
-
-		fileinfos = append(fileinfos, FileInfo{
-			IsDir:   f.IsDir(),
-			Name:    f.Name(),
-			Size:    f.Size(),
-			URL:     url.String(),
-			ModTime: f.ModTime(),
-			Mode:    f.Mode(),
-		})
-	}
-
-	return Listing{
-		Name:    path.Base(urlPath),
-		Path:    urlPath,
-		CanGoUp: canGoUp,
-		Items:   fileinfos,
-		Context: middleware.Context{
-			Root: http.Dir(root),
-			Req:  r,
-			URL:  r.URL,
-		},
-	}, nil
+        var fileinfos []FileInfo
+        var urlPath = r.URL.Path
+        for _, f := range files {
+                name := f.Name()
+
+                // Directory is not browsable if it contains index file
+                for _, indexName := range IndexPages {
+                        if name == indexName {
+                                return Listing{}, errors.New("Directory contains index file, not browsable!")
+                        }
+                }
+
+                // Ignore Caddyfile
+                if strings.ToLower(name)=="caddyfile" {
+                        continue
+                }
```

Comment:
Ah, this must be the change. (Next time, run go fmt to keep the formatting consistent.)